internal/log_analysis/log_processor/sources/s3_client_cache.go

```diff
@@ -190,11 +195,37 @@ func getSourceInfo(s3Object *S3ObjectInfo) (*models.SourceIntegration, error) {
 		integrationBucket, integrationPrefix := getSourceS3Info(source)
 		if aws.StringValue(integrationBucket) == s3Object.S3Bucket {
 			if strings.HasPrefix(s3Object.S3ObjectKey, aws.StringValue(integrationPrefix)) {
-				return source, nil
+				result = source
+				break
 			}
 		}
 	}
-	return nil, nil
+
+	// If the incoming notification maps to a known source, update the source information
+	if result != nil {
+		previousTime := lastEventReceived[*result.IntegrationID]
+		if previousTime.Add(statusUpdateFrequency).Before(now) {
```

Comment:
```suggestion
		if time.Since(previousTime) > statusUpdateFrequency
```

Similarly, use `time.Since` on line 180. It reads much more clearly

Reply:
> Similarly, use `time.Since` on line 180. It reads much more clearly

Hmmm, I don't see a way to do this while using the `now` variable.

Reply:
Right, you don't need the `now` variable at all then

Reply:
I was trying to reuse the `now` value to avoid calling `time.Now` for a second time inside the same function (`time.Since` calls `time.Now` internally).
Apparently Go doesn't make a syscall every time `time.Now` is invoked (it seems it has a type of thread-local cached value) but it seems it is considered a good practice to avoid invoking many times in "hot paths". This path is accessed frequently, but not THAT much (max a few 10s per second)

I did some small refactoring and I think the code is now more readable while reusing the time value. If you think there are still readability issues, I will switch it.